src/libraries/Microsoft.Extensions.Hosting/src/HostApplicationBuilder.cs

```diff
@@ -0,0 +1,435 @@
+                return this;
+
+            }
+
+            public IHostBuilder UseServiceProviderFactory<TContainerBuilder>(Func<HostBuilderContext, IServiceProviderFactory<TContainerBuilder>> factory) where TContainerBuilder : notnull
+            {
+                _serviceProviderFactory = new ServiceFactoryAdapter<TContainerBuilder>(() => _hostApplicationBuilder._hostBuilderContext, factory ?? throw new ArgumentNullException(nameof(factory)));
+                return this;
+            }
+
+            public IHostBuilder ConfigureContainer<TContainerBuilder>(Action<HostBuilderContext, TContainerBuilder> configureDelegate)
+            {
+                _configureContainerActions.Add(new ConfigureContainerAdapter<TContainerBuilder>(configureDelegate
+                    ?? throw new ArgumentNullException(nameof(configureDelegate))));
+
+                return this;
+            }
+        }
+
+        internal sealed class CheckedServiceCollection : IServiceCollection
+        {
+            private readonly IServiceCollection _services = new ServiceCollection();
+
+            public bool IsReadOnly { get; set; }
+            public int Count => _services.Count;
+
+            public ServiceDescriptor this[int index]
+            {
+                get => _services[index];
+                set
+                {
+                    CheckReadOnly();
+                    _services[index] = value;
+                }
+            }
+
+            public IEnumerator<ServiceDescriptor> GetEnumerator() => _services.GetEnumerator();
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+            public int IndexOf(ServiceDescriptor item) => _services.IndexOf(item);
+            public bool Contains(ServiceDescriptor item) => _services.Contains(item);
+            public void CopyTo(ServiceDescriptor[] array, int arrayIndex) => _services.CopyTo(array, arrayIndex);
+
+            public void Add(ServiceDescriptor item)
+            {
+                CheckReadOnly();
+                _services.Add(item);
+            }
+
+            public void Clear()
+            {
+                CheckReadOnly();
+                _services.Clear();
+            }
+
+            public void Insert(int index, ServiceDescriptor item)
+            {
+                CheckReadOnly();
+                _services.Insert(index, item);
+            }
+
+            public bool Remove(ServiceDescriptor item)
+            {
+                CheckReadOnly();
+                return _services.Remove(item);
+            }
+
+            public void RemoveAt(int index)
+            {
+                CheckReadOnly();
+                _services.RemoveAt(index);
+            }
+
+            private void CheckReadOnly()
+            {
+                if (IsReadOnly)
+                {
+                    throw new InvalidOperationException("The service collection cannot be modified after the application is built.");
```

Comment:
Do I need to add this and the other exception messages to Strings.resx? It looks like most exception messages in this project are in there, but [not all.](https://github.com/dotnet/runtime/blob/67d8b96d6d04d8aae231425bbf2a192966c4a3f8/src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs#L42)

Reply:
Since there is a Strings.resx already in place, yes please add the new strings to it.